voicevox_engine/preset/PresetManager.py

```diff
@@ -0,0 +1,182 @@
+from pathlib import Path
+
+import yaml
+from pydantic import ValidationError
+
+from .Preset import Preset
+from .PresetError import PresetError
+
+
+class PresetManager:
+    def __init__(
+        self,
+        preset_path: Path,
+    ):
+        self.presets = []
+        self.last_modified_time = 0
+        self.preset_path = preset_path
+
+    def load_presets(self):
+        """
+        プリセットのYAMLファイルを読み込む
+
+        Returns
+        -------
+        ret: List[Preset]
+            プリセットのリスト
+        """
+        _presets = []
+
+        # 設定ファイルのタイムスタンプを確認
+        try:
+            _last_modified_time = self.preset_path.stat().st_mtime
+            if _last_modified_time == self.last_modified_time:
+                return self.presets
+        except OSError:
+            raise PresetError("プリセットの設定ファイルが見つかりません")
+
+        try:
+            with open(self.preset_path, encoding="utf-8") as f:
```

Comment:
必須ではないと思いますが、openにmode指定がほしそうです。
もしくは、`pathlib.Path`を使っているので、`open`ではなく`self.preset_path.read_text`を使うように変えてもいいかもです。


PresetManager.py内のほかのコードに記述を合わせると、こうでしょうか。

```suggestion
            with open(self.preset_path, mode="r", encoding="utf-8") as f:
```

Reply:
`mode="r"`を追加しました。
`read_text` / `write_text`については、`yaml.safe_load` / `yaml.safe_dump`が`Stream`を受け取るようになっていて、書き換えてもそれほど読みやすくなるわけではないと思ったため、そのままにしています。